DKImagePickerController/DKImagePickerController.swift

```diff
@@ -206,104 +212,105 @@ public class DKImagePickerController: UINavigationController {
-			camera.dismissViewControllerAnimated(true, completion: nil)
-		}
-		
-		camera.didFinishCapturingImage = { [unowned self] (image) in
-			var newImageIdentifier: String!
-			PHPhotoLibrary.sharedPhotoLibrary().performChanges(
-				{ () in
-					let assetRequest = PHAssetChangeRequest.creationRequestForAssetFromImage(image)
-					newImageIdentifier = assetRequest.placeholderForCreatedAsset!.localIdentifier
-				}, completionHandler: { (success, error) -> Void in
-					dispatch_async(dispatch_get_main_queue(), { () -> Void in
-						if success {
-							if let newAsset = PHAsset.fetchAssetsWithLocalIdentifiers([newImageIdentifier], options: nil).firstObject as? PHAsset {
-								self.dismissViewControllerAnimated(true, completion: nil)
-								self.selectedImage(DKAsset(originalAsset: newAsset), needsToDismiss: true)
-							}
-						} else {
-							self.dismissViewControllerAnimated(true, completion: nil)
-							self.selectedImage(DKAsset(image: image), needsToDismiss: true)
-						}
-					})
-			})
-		}
-		self.checkCameraPermission(camera)
-		
-		return camera
-	}
-	
-	internal func checkCameraPermission(camera: DKCamera) {
-		func cameraDenied() {
-			dispatch_async(dispatch_get_main_queue()) {
-				let permissionView = DKPermissionView.permissionView(.Camera)
-				camera.cameraOverlayView = permissionView
-			}
-		}
-		
-		func setup() {
-			camera.cameraOverlayView = nil
-		}
-		
-		DKCamera.checkCameraPermission { granted in
-			granted ? setup() : cameraDenied()
-		}
-	}
-	
-	internal func presentCamera() {
-		self.presentViewController(self.createCamera(), animated: true, completion: nil)
-	}
-	
-	internal func dismiss() {
-		self.dismissViewControllerAnimated(true, completion: nil)
-		self.didCancel?()
-	}
-	
+
+    private func createCamera() -> DKCamera {
+        let camera = DKCamera()
+        camera.use16to9Ratio = self.use16to9Ratio
+
+        camera.didCancel = {[unowned camera] () -> Void in
+            camera.dismissViewControllerAnimated(true, completion: nil)
+        }
+
+        camera.didFinishCapturingImage = { [unowned self] (image) in
+            var newImageIdentifier: String!
+            PHPhotoLibrary.sharedPhotoLibrary().performChanges(
+                { () in
+                    let assetRequest = PHAssetChangeRequest.creationRequestForAssetFromImage(image)
+                    newImageIdentifier = assetRequest.placeholderForCreatedAsset!.localIdentifier
+                }, completionHandler: { (success, error) -> Void in
+                    dispatch_async(dispatch_get_main_queue(), { () -> Void in
+                        if success {
+                            if let newAsset = PHAsset.fetchAssetsWithLocalIdentifiers([newImageIdentifier], options: nil).firstObject as? PHAsset {
+                                self.dismissViewControllerAnimated(true, completion: nil)
+                                self.selectedImage(DKAsset(originalAsset: newAsset), needsToDismiss: false)
```

Comment:
So the difference is that it won't dismiss the gallery picking view directly after capturing. Are you saying there's another mode where taking a picture is the sole usage? In that case we probably should add a condition here